### DISCUSSION TITLE
Overloading??

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#84] [Q&A] [answered]
```
( symbol test-symbol
    (int :x ==>)
    (
        x puts!
    )
) ::

( symbol test-symbol    ;This definition changes the unseal-symbol `test-symbol` with itself.
    (int :x int :y ==>)
    (
        x y get-stack puts! clear-stack
    )
) ::

3 5 test-symbol puts     ; (3 5)
3    test-symbol puts     ; <ERROR>
```

How can **overload** this symbols? Or do you know a good way to do same?

**Comment by ghost** [comment]
I don't want to create more files to reach same symbols and for using their different definitions.
For example:
```
... *file1/test-symbol
... *file2/test-symbol
```

**Comment by ghost** [comment]
I'll use **overloading** for **default parameter initialization**.

**Comment by h3rald** [comment]
Hmmm. I think you can't do overloading in a concatenative language, because you don't pass parameters to operators, they are taken from the stack based on the operator logic/signature...

Now, in case you have the same number of parameters something could be done, but it's not going to be easy. Not without changing the way signatures work and not without changing symbol lookup.

We need a clever idea for this... it would be cool, because that's the reason why `length` is used for strings and `size` is used for quotations... 😕

**Comment by ghost** [comment, marked as answer]
Look at -> #99
